genericjpa/db/src/main/java/org/hibernate/search/annotations/UpdateInfo.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.annotations;
+
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * @author Martin Braun
+ * @hsearch.experimental
+ */
+@Target({FIELD, METHOD, TYPE})
+@Retention(RUNTIME)
+@Repeatable(UpdateInfos.class)
+public @interface UpdateInfo {
```

Comment:
@gunnarmorling:

> You can obtain some information through the JPA meta-model (EntityManager#getMetamodel()), e.g. on id types. For other things (table name etc.) you  need to go provider-specific APIs. I don't think it'll be super-complicated; For Hibernate ORM we could give you the right starting point, for others you'd have to dig a bit. But essentially that's what I think should be done here. 

I already looked into that and I think you underestimate how much work this is (apart from Hibernate ORM).

> Personally, I'd push back the thought of generification for now and rather focus on delivering > a great async backend for Hibernate ORM. Then the solution can be grown from there, but I > wouldn't start with a less than optimal API as that's what drives users away and you'll have a > hard time to make them come back.

That's more or less what I intended to say. Hibernate Search (with ORM) gets its own approach, while the generic version stays back. I more or less intended this PR to be a starter to get things going.